crates/macro/src/html_tree/html_dashed_name.rs

```diff
@@ -0,0 +1,85 @@
+use crate::Peek;
+use boolinator::Boolinator;
+use proc_macro2::Ident;
+use proc_macro2::Span;
+use quote::{quote, ToTokens};
+use std::fmt;
+use syn::buffer::Cursor;
+use syn::parse::{Parse, ParseStream, Result as ParseResult};
+use syn::Token;
+
+pub struct HtmlDashedName {
+    pub name: Ident,
+    pub extended: Vec<(Token![-], Ident)>,
+}
+
+impl HtmlDashedName {
+    pub fn new(name: Ident) -> Self {
+        HtmlDashedName {
+            name,
+            extended: Vec::new(),
+        }
+    }
+}
+
+impl fmt::Display for HtmlDashedName {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.name)?;
+        for (_, ident) in &self.extended {
+            write!(f, "-{}", ident)?;
+        }
+        Ok(())
+    }
+}
+
+impl Peek<'_, Self> for HtmlDashedName {
+    fn peek(cursor: Cursor) -> Option<(Self, Cursor)> {
+        let (name, cursor) = cursor.ident()?;
+        (name.to_string().to_lowercase() == name.to_string()).as_option()?;
+
+        let mut extended = Vec::new();
+        let mut cursor = cursor;
+        loop {
+            if let Some((punct, p_cursor)) = cursor.punct() {
+                if punct.as_char() == '-' {
```

Comment:
Since spaces are ignored (at least that's how I understand proc macros), wouldn't something like this be accepted?

```html
<tag-name -with  -spaces id="foo" />
```

Reply:
Yes, that's valid